dss_datamover/socket_communication.py

```diff
@@ -325,6 +321,9 @@ def recv_json(self, format="JSON", timeout=RECV_TIMEOUT):
                 msg_len = int(msg_len_in_bytes.decode('utf8'))
         except socket.timeout as e:
             raise e
+        except BlockingIOError as e:
```

Comment:
Here also we need to add the above logic where the header and payload are received together similar to clientsocket